### ISSUE TITLE
Unable to start app in dev env

### ISSUE THREAD

**Issue by skwerlman** [boydm/scenic_driver_glfw#25]
Running the app generated by `mix scenic.new` without specifying the env (or specifying the `dev` env) results in the following error:
```
sk:/mnt/code/deck_ex $ mix scenic.run
sh: /mnt/code/deck_ex/_build/dev/lib/scenic_driver_glfw/priv/prod/scenic_driver_glfw: No such file or directory

19:53:33.183 [error] dirty close
```

running it using in the prod env works as expected:
```
sk:/mnt/code/deck_ex $ MIX_ENV=prod mix scenic.run
Compiling 1 file (.ex)

19:55:52.065 [info]  Received event: {:viewport_enter, {320.0, 239.0}}

19:55:52.076 [info]  Received event: {:cursor_enter, 4}

19:55:52.076 [info]  Received event: {:cursor_pos, {320.0, 239.0}}

19:55:53.272 [info]  Received event: {:cursor_pos, {319.0, 236.0}}

(cont)
```

running tests also seems to work, though no tests are defined

### version info
scenic: 0.10.2
scenic_driver_glfw: 0.10.2
elixir/erlang: 
```
Erlang/OTP 22 [erts-10.4.1] [source] [64-bit] [smp:32:32] [ds:32:32:10] [async-threads:1] [hipe]

Elixir 1.8.2 (compiled with Erlang/OTP 22)
```
linux (archlinux):
```
Linux a2 5.1.15-zen1-1-zen #1 ZEN SMP PREEMPT Tue Jun 25 04:49:28 UTC 2019 x86_64 GNU/Linux
```

lemme know if there's any more info you need

**Comment by boydm** [comment]
The dirty close indicates the driver had a crash somewhere. 

This is going to be hard for me to track down as I don't use or have archlinux installed. Probably a issue with the graphics driver.

Which driver are you using? Also, is Wayland installed?

**Comment by skwerlman** [comment]
I'm using the proprietary nvidia driver (the dkms variant), version 430.26-7 along with glew 2.1.0-1 and glfw-x11 3.3-2

wayland is installed, along with xorg-server-xwayland; however, using the wayland variants of glew and glfw im unable to get it to work even in prod

**Comment by axelson** [comment]
Hmmm, this looks like this might be the same issue I'm running into in https://github.com/boydm/scenic_new/issues/36 (and I'm also on Arch Linux). For me I've noticed that it works if I run:
 ```
rm -rf _build/dev/lib/scenic_driver_glfw
mix scenic.run
```

Also the main difference appears to be that `_build/dev/lib/scenic_driver_glfw/priv` is a symlink after running the commands above, while initially it is not a symlink.

**Comment by axelson** [comment]
As far as I can tell the issue seems to be that when the environment variable `MIX_ENV` is not set or is set to `dev` that `Mix.env()` is returning `:prod` which messes up all the paths. But I have been unable to track down why `Mix.env()` is returning an incorrect value.

**Comment by skwerlman** [comment]
can confirm this workaround allows me to start the app in `:dev`

**Comment by axelson** [comment]
Also it looks like it is expected that dependencies are compiled with mix env `:prod`: https://hexdocs.pm/mix/Mix.Tasks.Deps.html#module-dependency-definition-options

**Comment by nyaray** [comment]
I can confirm that setting `MIX_ENV` to `prod` works to get a newly generated app to start on mac.

**Comment by axelson** [comment]
Hmmm, setting `MIX_ENV` to `prod` doesn't workaround the issue for me.

**Comment by axelson** [comment]
This can be closed now that #26 is fixed (unless it shouldn't be closed until the next release is cut)

**Comment by boydm** [comment]
The fix for this is now published to hex in v0.10.1

**Comment by skwerlman** [comment]
fix is confirmed working on a new project (and in previously broken ones)! thanks!
